src/test/java/com/marklogic/kafka/connect/source/ConstraintInjectionTest.java

```diff
@@ -0,0 +1,54 @@
+package com.marklogic.kafka.connect.source;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class ConstraintInjectionTest extends AbstractIntegrationSourceTest {
+
+    @Test
+    void testConstraintInjection() {
+        String originalDsl = "op.fromView(\"Medical\", \"Authors\")";
```

Comment:
I think these should all be separate unit tests, each with a good name that identifies the scenario they're testing. That also allows each one to succeed/fail in isolation and makes each one easier to understand. 

Also, I like how the "max value" test is also exercising the query - I think that's necessary here too. It's good to verify that the query is exactly as we expect it, but the test should also execute the query to verify that we get back the expected number of rows.

Reply:
Well, this sucks.
Limit doesn't work here. I'm assuming the problem is with RowBatcher.

Reply:
Resolved with: MarkLogicSourceConfig.DMSDK_BATCH_SIZE = String.valueOf(Integer.MAX_VALUE)